main.go

```diff
@@ -190,10 +216,17 @@ func computeRoutePrefix(prefix string, externalURL *url.URL) string {
 	return prefix
 }
 
-func interruptHandler(l net.Listener, logger log.Logger) {
+func exitHandler(l net.Listener, quitCh <-chan struct{}, logger log.Logger) {
```

Comment:
`Handler` is kind of confusing as a name because this is not an HTTP handler as above. It was already confusing prior to this PR, but one could have argued that "signal handling" is common Unix lingo. Now that it is handling more than a signal, that argument doesn't hold anymore. How about `closeListenerOnQuit` or something, and add a doc comment, because the function is doing two things now, which is harder to see at a first glance. Something like "closeListenerOnQuite closes the provided listener upon closing the provided quitCh or upon receiving a SIGTERM."

Reply:
I had similar feelings about this one initially :smile:

Reply:
I just added SIGINT signal to your proposed comment... I even start ending my sentences into github comments with periods :smile: .